tests/sweep_framework/sweeps/line_all_gather.py

```diff
@@ -0,0 +1,178 @@
+# Each suite has a key name (in this case "suite_1" and "suite_2") which will associate the test vectors to this specific suite of inputs.
+# Developers can create their own generator functions and pass them to the parameters as inputs.
+parameters = {
+    "suite_1": {
+        "num_devices": [4, 8],
+        "num_links": [1, 2],
+        "input_shape": [
+            [8, 1, 33, 256],
+            [8, 1, 256, 32],
+            [8, 8, 256, 384],
+            [8, 5, 13, 512],
+            [8, 5, 32, 512],
+            [1, 1, 32, 16384],
+        ],
+        "dim": [0, 1, 3],
+        "layout": [ttl.tensor.Layout.ROW_MAJOR, ttl.tensor.Layout.TILE],
+        "input_dtype": [ttl.tensor.DataType.BFLOAT16],
+        "mem_config": [ttl.tensor.MemoryConfig(buffer_type=ttl.tensor.BufferType.DRAM)],
+        "enable_async": [True, False],
+    },
+}
+
+
+# Invalidate vector is called during the generation phase where each vector will be passed in.
+# If invalidated, the vector will still be stored but will be skipped.
+# Returns False, None if the vector is valid, and True, str with a reason for invalidation if it is invalid.
+def invalidate_vector(test_vector) -> Tuple[bool, Optional[str]]:
+    if test_vector["broadcast"] in {"w", "hw"} and test_vector["input_b_layout"] == ttnn.ROW_MAJOR_LAYOUT:
+        return True, "Broadcasting along width is not supported for row major layout"
+    return False, None
+
+
+def skip(
+    *, all_devices, input_shape, dim, mem_config, num_devices, num_links, input_dtype, layout, **_
+) -> Tuple[bool, Optional[str]]:
+    if len(all_devices) != 8:
+        return True, "Not T3000!"
+    (is_known_failure, message) = is_unsupported_case(
+        input_shape, dim, mem_config, num_devices, num_links, input_dtype, layout
+    )
+    if is_known_failure:
+        return True, f"Skipping unsupported case {message}."
+    return False, None
+
+
+# This is the run instructions for the test, defined by the developer.
+# The run function must take the above-defined parameters as inputs.
+# The runner will call this run function with each test vector, and the returned results from this function will be stored.
+def run(
+    all_devices,
+    num_devices,
+    input_shape,
+    dim,
+    num_links,
+    input_dtype,
+    layout,
+    mem_config,
+    use_program_cache,
+    function_level_defaults,
+    enable_async,
+    num_iters=1,
+) -> list:
+    for device in all_devices:
+        device.enable_async(enable_async)
+
+    logger.info(f"Input shape: {input_shape}")
+    logger.info(f"dim: {dim}")
+
+    devices = get_devices_for_t3000(all_devices, num_devices)
+    # for device in devices:
+    #    device.disable_and_clear_program_cache()
+
+    logger.info(f"Input shape: {input_shape}")
+    logger.info(f"dim: {dim}")
+
+    input_tensor = torch.rand(input_shape).bfloat16()
+
+    input_tensors = torch.chunk(input_tensor, num_devices, dim)
```

Comment:
This is not using the proper `ShardTensor2dMesh` API as outlined in this PR: https://github.com/tenstorrent/tt-metal/pull/10872. Please update this and all future tests to use this API as it's the official recommendation way to describe these tensors now.